crates/bevy_animation/src/animation_curves.rs

```diff
@@ -0,0 +1,544 @@
+///     #[derive(Reflect)]
+///     struct FontSizeProperty;
+///
+///     impl AnimatableProperty for FontSizeProperty {
+///         type Component = Text;
+///         type Property = f32;
+///         fn get_mut(component: &mut Self::Component) -> Option<&mut Self::Property> {
+///             Some(&mut component.sections.get_mut(0)?.style.font_size)
+///         }
+///     }
+///
+/// You can then create an [`AnimationClip`] to animate this property like so:
+///
+///     # use bevy_animation::{AnimationClip, AnimationTargetId, VariableCurve};
+///     # use bevy_animation::prelude::{AnimatableProperty, AnimatableKeyframeCurve, AnimatableCurve};
+///     # use bevy_core::Name;
+///     # use bevy_reflect::Reflect;
+///     # use bevy_text::Text;
+///     # let animation_target_id = AnimationTargetId::from(&Name::new("Test"));
+///     # #[derive(Reflect)]
+///     # struct FontSizeProperty;
+///     # impl AnimatableProperty for FontSizeProperty {
+///     #     type Component = Text;
+///     #     type Property = f32;
+///     #     fn get_mut(component: &mut Self::Component) -> Option<&mut Self::Property> {
+///     #         Some(&mut component.sections.get_mut(0)?.style.font_size)
+///     #     }
+///     # }
+///     let mut animation_clip = AnimationClip::default();
+///     animation_clip.add_curve_to_target(
+///         animation_target_id,
+///         AnimatableKeyframeCurve::new(
+///             [
+///                 (0.0, 24.0),
+///                 (1.0, 80.0),
+///             ]
+///         )
+///         .map(AnimatableCurve::<FontSizeProperty, _>::from_curve)
+///         .expect("Failed to create font size curve")
+///     );
+///
+/// Here, the use of `AnimatableKeyframeCurve` creates a curve out of the given keyframe time-value
+/// pairs, using the `Animatable` implementation of `f32` to interpolate between then. The
+/// invocation of [`AnimatableCurve::from_curve`] with `FontSizeProperty` indicates that the `f32`
+/// output from that curve is to be used to animate the font size of a `Text` component (as
+/// configured above).
+///
+/// [`AnimationClip`]: crate::AnimationClip
+pub trait AnimatableProperty: Reflect + TypePath {
+    /// The type of the component that the property lives on.
+    type Component: Component;
+
+    /// The type of the property to be animated.
+    type Property: Animatable + FromReflect + Reflectable + Clone + Sync + Debug;
+
+    /// Given a reference to the component, returns a reference to the property.
+    ///
+    /// If the property couldn't be found, returns `None`.
+    fn get_mut(component: &mut Self::Component) -> Option<&mut Self::Property>;
+}
+
+/// This trait collects the additional requirements on top of [`Curve<T>`] needed for a
+/// curve to be used as an [`AnimationCurve`].
+pub trait InnerAnimationCurve<T>: Curve<T> + Debug + Clone + Reflectable {}
+
+impl<T, C> InnerAnimationCurve<T> for C where C: Curve<T> + Debug + Clone + Reflectable {}
```

Comment:
Not really; mostly this is just a helpful grouping of constraints to avoid repeating all these trait bounds, but even in principle, `T` could still basically be anything, as long as the `AnimationCurve` implementation knows how to turn its output into a component mutation.